lib/Scheduling/Problems.cpp

```diff
@@ -407,6 +407,37 @@ LogicalResult ModuloProblem::verify() {
   return success();
 }
 
+//===----------------------------------------------------------------------===//
+// ChainingCyclicProblem
+//===----------------------------------------------------------------------===//
+
+LogicalResult ChainingCyclicProblem::checkSSADep(Dependence dep) {
+  if (!dep.isAuxiliary() && (this->getDistance(dep).value_or(0) != 0))
+    return getContainingOp()->emitError()
+           << "Non-zero ssa dependence not allowed.\n"
```

Comment:
Missing a "distance" somewhere, maybe: "Def-use dependence cannot have a non-zero distance" or so.

Reply:
Please add an error test-case for this check, then.